packages/teams-js/src/public/thirdPartyCloudStorage.ts

```diff
@@ -0,0 +1,236 @@
+    /** A string representing the MIME type of the file */
+    fileType: string;
+    /** An array of {@link AssembleAttachment | AssembleAttachment} objects representing files to be sent as attachment */
+    assembleAttachment: AssembleAttachment[];
+  }
+
+  /**
+   * Defines the Callback function received from Third Party App
+   * @beta
+   */
+  export interface DragAndDropFileCallback {
+    /** Callback from third party app */
+    (files: FileFor3PApps[], error?: SdkError): void;
+  }
+
+  let callback: DragAndDropFileCallback | null = null;
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param {string} dragAndDropInput - thread id/ conversation id of the chat/channel received from the app.
+   * @param {DragAndDropFileCallback} dragAndDropFileCallback - callback
+   *   A callback function to handle the result of the operation
+   * @beta
+   */
+  export function getDragAndDropFiles(
+    dragAndDropInput: string,
+    dragAndDropFileCallback: DragAndDropFileCallback,
+  ): void {
+    if (!dragAndDropFileCallback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+    if (!dragAndDropInput || dragAndDropInput === '') {
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      dragAndDropFileCallback([], invalidInput);
+      return;
+    }
+
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    if (callback) {
+      callback = null;
+      throw new Error('getDragAndDropFiles cannot be called twice');
+    } else {
+      callback = dragAndDropFileCallback;
+    }
+
+    sendMessageToParent(
+      'thirdPartyCloudStorage.getDragAndDropFiles',
+      [dragAndDropInput],
+      handleGetDragAndDropFilesCallbackRequest,
+    );
+  }
+
+  function handleGetDragAndDropFilesCallbackRequest(fileResult: FileResult): void {
+    if (callback) {
+      if (fileResult && fileResult.error) {
+        callback([], fileResult.error);
+        callback = null;
+      } else {
+        if (fileResult && fileResult.fileChunk) {
+          try {
+            const assemble: AssembleAttachment | null = decodeAttachment(fileResult.fileChunk, fileResult.fileType);
+            if (assemble) {
+              helper.assembleAttachment.push(assemble);
+            } else {
+              Files3PLogger(
+                `Received a null assemble attachment for when decoding chunk sequence ${fileResult.fileChunk.chunkSequence}; not including the chunk in the assembled file.`,
+              );
+              callback([], { errorCode: ErrorCode.INTERNAL_ERROR, message: 'error occurred while receiving data' });
+              files = [];
+              callback = null;
+            }
+
+            // we will send the maximum integer as chunkSequence to identify the last chunk
+            if (fileResult.fileChunk.chunkSequence === Number.MAX_SAFE_INTEGER) {
```

Comment:
Is `MAX_SAFE_INTEGER` the same value on every platform/language combination? A more durable solution would be to store the count of chunks on FileResult (although Vaibhav said that would require extra processing you aren't doing now)

Reply:
This value needs to be sent by host app, if not Web app then pass 2^53-1 value.